### PR TITLE
[ET-957] Add warning if no commerce provider available

### PR DESCRIPTION
**Pull request by rafsuntaskin**
[ET-957]

Screenshot:

Screen Shot 2020-11-30 at 6 43 23 PM

[ET-957]: https://moderntribe.atlassian.net/browse/ET-957

### CODE REVIEW

File: src/Tribe/Commerce/Warnings.php [hunk 1]
```diff
@@ -39,6 +40,24 @@ public function recurring_event_warning_msg( $post_id ) {
 		$this->show_notice( $warning );
 	}
 
+	/**
+	 * Add Provider missing warning for tickets.
```
Comment (tr1b0t) [inline src/Tribe/Commerce/Warnings.php:44]: :no_entry_sign: Doc comment for parameter "$post_id" missing

<sub>hash: e07416325bcd02b11d397b544c332b12</sub>

File: src/Tribe/Commerce/Warnings.php [hunk 2]
```diff
@@ -39,6 +40,24 @@ public function recurring_event_warning_msg( $post_id ) {
...
+	 *
+	 * @param $post_id
+	 */
```
Comment (tr1b0t) [inline src/Tribe/Commerce/Warnings.php:47]: :no_entry_sign: Missing parameter name

<sub>hash: a37e520dea8a6a8f16ef02d345ae82f7</sub>

File: src/Tribe/Commerce/Warnings.php [hunk 3]
```diff
@@ -39,6 +40,24 @@ public function recurring_event_warning_msg( $post_id ) {
...
+	public function add_commerce_provider_warning( $post_id ) {
+		$available_modules = array_diff_key( Tribe__Tickets__Tickets::modules(), [ 'Tribe__Tickets__RSVP' => true ] );
+
+		if ( count( $available_modules ) > 0 ) {
+			return;
+		}
+
+		$kb_link = 'http://m.tri.be/1ao5';
+		$message = sprintf( __( 'There is no payment gateway configured. To create tickets, you\'ll need to enable and configure an ecommerce solution. <a href="%1$s" target="_blank" rel="noopener noreferrer">[Learn more]</a>', 'event-tickets' ), $kb_link );
+
```
Comment (tr1b0t) [inline src/Tribe/Commerce/Warnings.php:57]: :no_entry_sign: A gettext call containing placeholders was found, but was not accompanied by a "translators:" comment on the line above to clarify the meaning of the placeholders.

<sub>hash: cc85894db7d445bd10d3f1316f7262b5</sub>

**Review by tr1b0t — COMMENTED** [review]
:no_entry_sign: 3 error(s)

**Review by sc0ttkclark — CHANGES_REQUESTED** [review]
Some valid bot problems to resolve here, needs changelog

File: src/Tribe/Editor/Warnings.php [hunk 4]
```diff
@@ -46,6 +47,40 @@ public function show_recurring_event_warning_message( $post_id ) {
...
+
+	/**
+	 * Get the Commerce Provider missing warning message.
+	 *
+	 * @since TBD
+	 *
+	 * @return string The Commerce Provider missing message.
+	 */
+	public function get_commerce_provider_missing_warning_message() {
+		$kb_url      = 'http://m.tri.be/1ao5';
```
Comment (tr1b0t) [inline src/Tribe/Editor/Warnings.php:73]: :warning: Equals sign not aligned correctly; expected 1 space but found 6 spaces

<sub>hash: 00c9e53f0c179bf52e50a7080f21655e</sub>

File: src/Tribe/Editor/Warnings.php [hunk 5]
```diff
@@ -46,6 +47,40 @@ public function show_recurring_event_warning_message( $post_id ) {
...
+
+		$link = sprintf(
+				'<a href="%1$s" target="_blank" rel="noopener noreferrer">%2$s</a>',
```
Comment (tr1b0t) [inline src/Tribe/Editor/Warnings.php:76]: :no_entry_sign: Multi-line function call not indented correctly; expected 12 spaces but found 16

<sub>hash: ffd46ae0d4f404755e4da9e4d397a8ce</sub>

File: src/Tribe/Editor/Warnings.php [hunk 6]
```diff
@@ -46,6 +47,40 @@ public function show_recurring_event_warning_message( $post_id ) {
...
+				esc_url( $kb_url ),
```
Comment (tr1b0t) [inline src/Tribe/Editor/Warnings.php:77]: :no_entry_sign: Multi-line function call not indented correctly; expected 12 spaces but found 16

<sub>hash: 482dfa51b22579e5cbfdae5637d1955e</sub>

File: src/Tribe/Editor/Warnings.php [hunk 7]
```diff
@@ -46,6 +47,40 @@ public function show_recurring_event_warning_message( $post_id ) {
...
+				esc_html_x( '[Learn More]', 'Helper link in Ticket Editor', 'event-tickets' )
```
Comment (tr1b0t) [inline src/Tribe/Editor/Warnings.php:78]: :no_entry_sign: Multi-line function call not indented correctly; expected 12 spaces but found 16

<sub>hash: 517f9434919a58b80b792d20f6ac89d0</sub>

File: src/Tribe/Editor/Warnings.php [hunk 8]
```diff
@@ -46,6 +47,40 @@ public function show_recurring_event_warning_message( $post_id ) {
...
+		);
+
+		return sprintf( __( 'There is no payment gateway configured. To create tickets, you\'ll need to enable and configure an ecommerce solution. %1$s', 'event-tickets' ), $link );
```
Comment (tr1b0t) [inline src/Tribe/Editor/Warnings.php:81]: :no_entry_sign: A gettext call containing placeholders was found, but was not accompanied by a "translators:" comment on the line above to clarify the meaning of the placeholders.

<sub>hash: 330dfc66042566d0cb543ca1f83ec305</sub>

**Review by tr1b0t — COMMENTED** [review]
:no_entry_sign: 4 error(s)
:warning: 1 warning(s)

File: src/Tribe/Editor/Warnings.php [hunk 9]
```diff
@@ -46,6 +47,42 @@ public function show_recurring_event_warning_message( $post_id ) {
...
+
+		/* translators: %1$s: URL for help link, %2$s: Label for help link. */
+		$link = sprintf(
+			'<a href="%1$s" target="_blank" rel="noopener noreferrer">%2$s</a>',
+			esc_url( $kb_url ),
+			esc_html_x( '[Learn More]', 'Helper link in Ticket Editor', 'event-tickets' )
+		);
+
+		/* translators: %1$s: link to help article. */
+		return sprintf( __( 'There is no payment gateway configured. To create tickets, you\'ll need to enable and configure an ecommerce solution. %1$s', 'event-tickets' ), $link );
```
Comment (sc0ttkclark) [inline src/Tribe/Editor/Warnings.php:83]:
```suggestion
		return wp_kses_post(
			sprintf(
				/* translators: %1$s: link to help article. */
				__( 'There is no payment gateway configured. To create tickets, you\'ll need to enable and configure an ecommerce solution. %1$s', 'event-tickets' ),
				$link
			)
		);
```

**Review by sc0ttkclark — APPROVED** [review]

**Pull request merged**